### ISSUE TITLE
Please add ReadOnlySequence<byte> Overloads

### ISSUE THREAD

**Issue by AceHack** [dotnet/corefx#32497]
```
System.Net.Sockets.Socket
{
    public ValueTask<int> SendAsync(ReadOnlySequence<byte> buffer, SocketFlags socketFlags, CancellationToken cancellationToken = default(CancellationToken))
}

System.Text.Encoding
{
    public string GetString(ReadOnlySequence<byte> bytes)
}
```

See this gist for example usage.

https://gist.github.com/AceHack/786e558f14b2d9c0d46a1ea330c158f7

**Comment by davidfowl** [comment]
Agreed we should go through and look at doing this for the most common scenaros in 3.0.

cc @KrzysztofCwalina @ahsonkhan @stephentoub

**Comment by BrennanConroy** [comment]
`BitConverter.ToString` as well

**Comment by tarekgh** [comment]
@AceHack would you create a complete proposal? you may have a look at https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md for more info.

**Comment by davidfowl** [comment]
@tarekgh I can do the proposal but I have a question should I make a proposal per new API or should there be an uber proposal like the Span version of this. I don't think we'll need support for ROS in all the places we have span support but there's more than one.

**Comment by GrabYourPitchforks** [comment]
I removed the `Encoding` extension method API request from this issue because it's now being tracked by https://github.com/dotnet/corefx/issues/41166.

**Comment by karelz** [comment]
Triage:
In general we are ok with it.
We should consider also `Receive` and all other APIs (`SendTo`, `SendtoAsync`, etc., etc.).
We should double check that there is perf benefit.

@scalablecory floats around idea to use `Span<Memory<byte>>` that would work also for smaller buffers.

@davidfowl do you plan to work on the proposal and prototype to confirm perf benefit (e.g. on Kestrel some benchmark)?

**Comment by davidfowl** [comment]
> @scalablecory floats around idea to use Span<Memory<byte>> that would work also for smaller buffers.

How would that work? It's not like you could stackalloc. Also this method is async so the implementation would need to synchronously copy to something on the heap just in case the call goes async. What am I missing?


> @davidfowl do you plan to work on the proposal and prototype to confirm perf benefit (e.g. on Kestrel some benchmark)?

What do you mean here? Would I add the API and test it out in Kestrel?
